lucyparsons/OpenOversight
# Add reCAPTCHA and optional disable flag for user registration

## Status

Ready for review

## Description of Changes

This allows you to completely disable user registration by setting the environment variable `DISABLE_REGISTRATION`. Additionally, you can add Google reCAPTCHA v3 to the user registration page by setting the `RECAPTCHA_V3_SITE_KEY`, `RECAPTCHA_V3_SECRET_KEY`, and optionally `RECAPTCHA_V3_THRESHOLD` environment variables.

## To Test

Set up a recaptcha v3 site with Google (using localhost and 127.0.0.1 for domain names) and add the site key and secret key to the `.env` file using above environment variables. Then `make dev` and register a user. You can then see the recaptcha score assigned to that registration by checking the docker logs. You should see something like this:

```
[2020-07-20 18:14:31,903] INFO in recaptcha3: {'success': True, 'challenge_ts': '2020-07-20T18:14:17Z', 'hostname': 'localhost', 'score': 0.9, 'action': 'registration'}
```

## Notes for Deployment

## Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/2007008/86859121-f0bd1000-c08f-11ea-8b79-d9abd99b336e.png)


## Tests and linting

 - [x] I have rebased my changes on current `develop`

 - [x] pytests pass in the development environment on my local machine

 - [x] `flake8` checks pass
